Ignore backticks in documentation abstracts

natecook1000 opened the pull request:
The linguistic tagger doesn't do a great job recognizing backticks as opening vs. closing quotes, so some valid abstracts that contain backticks were being rejected by the `BeginDocumentationCommentWithOneLineSummary` rule. This change removes any `InlineCode` nodes from the abstract paragraph before processing, since they aren't important to that judgment.

Fixes #924.

ahoppen reviewed (commented):
Does this cause issues when you have a code block that contains a period and would thus recognize it as a multi-sentence first paragraph?

Eg.
```
/// The same as `Foo.foo()`.
func bar() {}
```

If not, could you add a test case for that?

allevato commented:
We can assume that a sentence-ending period will never occur _inside_ an inline code block, so I wonder if we should simply remove the content within backticks before doing this check, instead of removing just the backticks.

Since the `briefSummary` field is already a `Paragraph` node, we could do this in a structured way by just locally mutating that tree to remove any `InlineCode` children (and likewise for `SymbolLink` children, to handle the DocC-style double-backtick syntax).

The only thing I'm not sure about is how the linguistic tagger would handle a space before the period, because I'm assuming that removing the `InlineCode` node from this:

```
Foo bar `baz`.
```

would leave us with `"Foo bar ."`, and I'm not familiar enough with that API to know if it would handle that. If that's a problem, then we could address that by instead of removing the `InlineCode/SymbolLink` children, replacing them with a plaintext word like "placeholder", and then run the linguistic tagger on that result.

natecook1000 commented:
I would have sworn I updated this commit before opening the PR, but here we are. @allevato, I switched to use a `MarkupRewriter` to remove `InlineCode` nodes, and it seems to work great! I added a test case using a double-backtick link (which I don't think should have periods, but just in case) and it seemed to do fine.
